### DISCUSSION TITLE
Location (lat/long)?

### DISCUSSION THREAD

**Discussion by plettb** [SixLabors/ImageSharp#2254] [Q&A]
I have no trouble extracting an image's location using the ExifProfile provided by this library.  I'm just wondering:  are there any extension methods or helpers that might convert Rational[] to a decimal, or, better yet, go straight from the ExifProfile to NetTopologySuite.Geometries.Geometry?  This is obviously trivial code to write, but I try to stay away from reinventing the wheel. :)

**Comment by JimBobSquarePants** [comment]
You mean converting each individual item in the array yeah?

`Rational` already contains [`FromDouble(...)`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Rational.html#SixLabors_ImageSharp_Rational_FromDouble_System_Double_) and [`ToDouble(`)](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Rational.html#SixLabors_ImageSharp_Rational_ToDouble) methods.

**Reply by plettb** [reply]
Yes, `Rational` contains `ToDouble()`, but I was thinking there might be more (e.g. handling the entire array).  For example, here's a helper method I slapped together.  It's far from perfect, but it's a start.  Something like this would be very useful as an extension method or something.  I'm guessing many people have written similar code, and as I said, I'm a little loath to reinvent the wheel - that's what NuGet libraries are for. :)

```
Geometry? GetLocation(ExifProfile exifProfile)
{
    Geometry? location = null;
    if (exifProfile != null)
    {
        try
        {
            var latitudeArray = exifProfile.GetValue(ExifTag.GPSLatitude);
            if (latitudeArray != null)
            {
                var latitude = (double)latitudeArray.Value[0].Numerator / (double)latitudeArray.Value[0].Denominator +
                    ((double)latitudeArray.Value[1].Numerator / (double)latitudeArray.Value[1].Denominator) / 60 +
                    ((double)latitudeArray.Value[2].Numerator / (double)latitudeArray.Value[2].Denominator) / 3600;
                var latitudeRef = exifProfile.GetValue(ExifTag.GPSLatitudeRef)?.Value;
                if ((latitudeRef != null) && (!latitudeRef.Equals("N", StringComparison.OrdinalIgnoreCase)))
                    latitude = -latitude;

                var longitudeArray = exifProfile.GetValue(ExifTag.GPSLongitude);
                if (longitudeArray != null)
                {
                    var longitude = (double)longitudeArray.Value[0].Numerator / (double)longitudeArray.Value[0].Denominator +
                        ((double)longitudeArray.Value[1].Numerator / (double)longitudeArray.Value[1].Denominator) / 60 +
                        ((double)longitudeArray.Value[2].Numerator / (double)longitudeArray.Value[2].Denominator) / 3600;
                    var longitudeRef = exifProfile.GetValue(ExifTag.GPSLongitudeRef)?.Value;
                    if ((longitudeRef != null) && (!longitudeRef.Equals("E", StringComparison.OrdinalIgnoreCase)))
                        longitude = -longitude;

                    if ((latitude != 0) && (longitude != 0))
                        location = _geometryFactory.CreatePoint(new Coordinate(longitude, latitude));
                }
            }
        }
        catch (Exception)
        {
            // ignore
        }
    }

    return location;
}
```

P.S. If nothing like this currently exists, please feel free to code-review the above. :)

**Reply by JimBobSquarePants** [reply]
That's really out of the scope of ImageSharp. We capture the raw detail and it's up to libraries utilizing our APIs to add convenience methods to convert the basic types to their custom types. 

PS. Empty `catch` clauses are the devil

**Reply by plettb** [reply]
Yeah, I understand it may be out of scope for ImageSharp.  I was wondering if there were related libraries.

Also, I realize that `NetTopologySuite.Geometries.Geometry` may be too specific, but some kind of method that simply returns lat/long as doubles would be nice (or even a "ToDouble()" on "Rational[]"). :)

**Reply by JimBobSquarePants** [reply]
Nothing related I'm afraid. Simply too much work involved given the number of committers to do interop libraries. 

A Rational array wouldn't be cast to a single double converting an array to double is as simple as 
```c#
double[] doubles = rationals.Select(x => x.ToDouble()).ToArray();
```
